editor.planx.uk/src/routes/views/flowEditor.tsx

```diff
@@ -121,13 +130,23 @@ export const flowEditorView = async (req: NaviRequest) => {
   const {
     id,
     flowStatus,
-    flowAnalyticsLink,
     isFlowPublished,
     isTemplate,
     templatedFrom,
     template,
   } = await getFlowEditorData(flow, req.params.team);
 
+  let flowAnalyticsLink: string | null = null;
+
+  if (flowStatus === "online") {
+    flowAnalyticsLink =
+      generateAnalyticsLink({
+        environment,
+        flowId: id,
+        flowSlug: flow,
+      }) ?? null;
+  }
```

Comment:
```suggestion
const flowAnalyticsLink = generateAnalyticsLink({
  environment,
  flowId: id,
  flowSlug: flow,
  flowStatus,
});
```

I'd suggest grouping all this logic together. Currently, we have a flow status check and fallback to `null` outside the `generateAnalyticsLink()` function.

If we needed to call this again from somewhere, we'd need to synchronise this logic.